src/rezplugins/shell/_utils/powershell_base.py

```diff
@@ -153,10 +153,10 @@ def _record_shell(ex, files, bind_rez=True, print_msg=False):
         # only the bool $? var is set.
         #
         executor.command(
-            "if(! $? -or $LASTEXITCODE) {\n"
-            "  if ($LASTEXITCODE) {\n"
-            "    exit $LASTEXITCODE\n"
-            "  }\n"
+            "if ((Test-Path variable:LASTEXITCODE) -and $LASTEXITCODE) {\n"
```

Comment:
Can you add a comment here to explain what's going on? It would greatly increase the maintainability of this part of the code.

Reply:
I updated the existing comment with a bit more context, including a note about the semantics of strict mode. Let me know what you think.

Reply:
That's perfect, thanks!